lib/facter/powershell_version.rb

```diff
@@ -0,0 +1,12 @@
+require 'win32/registry'
+
+Facter.add(:powershell_version) do
+  confine :operatingsystem => :windows
+  setcode do
+    # HKLM:\SOFTWARE\Microsoft\PowerShell\3\PowerShellEngine
+    # Win32::Registry::HKEY_LOCAL_MACHINE.open('SOFTWARE\Microsoft\PowerShell\3\PowerShellEngine') do |reg|
+    #   reg_val = reg['PowerShellVersion']
+    # end
```

Comment:
If it's dead yes, but this was commented out to figure out why confine wasn't working. We are actually going to have to figure out the version on the client, so we need this.